libs/cardano-ledger-api/test/Test/Cardano/Ledger/Api/State/Imp/QuerySpec.hs

```diff
@@ -0,0 +1,314 @@
+    expectMembers $ Set.fromList [c1, c3, c4, c6, c7]
+    expectNoFilterQueryResult $
+      Just $
+        Map.fromList
+          [ (c1, CommitteeMemberState (MemberAuthorized (KeyHashObj hk1)) Active (Just 13) NoChangeExpected)
+          , (c3, CommitteeMemberState MemberResigned Active (Just 7) NoChangeExpected)
+          , (c4, CommitteeMemberState MemberNotAuthorized Expired (Just 4) NoChangeExpected)
+          , (c6, CommitteeMemberState (MemberAuthorized (KeyHashObj hk6)) Active (Just 6) ToBeExpired)
+          , (c7, CommitteeMemberState MemberNotAuthorized Active (Just 7) NoChangeExpected)
+          ]
+    expectQueryResult
+      Set.empty
+      Set.empty
+      (Set.fromList [Unrecognized])
+      (Just Map.empty)
+
+    -- elect new committee to be:
+    -- c4 (which is presently Expired, set a new term),
+    -- c6 (which is presently ToBeExpired, set a new term)
+    -- c7 (which will become ToBeExpired in the next epoch)
+    -- c3 (which would become ToBeExpired in the next epoch, but set a new term)
+    _ <-
+      electCommittee
+        (SJust ga2)
+        drep
+        (Set.fromList [c1])
+        ( Map.fromList
+            [ (c3, 9)
+            , (c4, 9)
+            , (c6, 9)
+            ]
+        )
+    passEpoch -- epoch 7
+    -- members whose term changed have next expoch change TermAdjusted
+    expectNoFilterQueryResult $
+      Just $
+        Map.fromList
+          [ (c1, CommitteeMemberState (MemberAuthorized (KeyHashObj hk1)) Active (Just 13) ToBeRemoved)
+          , (c3, CommitteeMemberState MemberResigned Active (Just 7) (TermAdjusted 9))
+          , (c4, CommitteeMemberState MemberNotAuthorized Expired (Just 4) (TermAdjusted 9))
+          , (c6, CommitteeMemberState (MemberAuthorized (KeyHashObj hk6)) Expired (Just 6) (TermAdjusted 9))
+          , (c7, CommitteeMemberState MemberNotAuthorized Active (Just 7) ToBeExpired)
+          ]
+    passEpoch -- epoch 8
+    expectMembers $ Set.fromList [c3, c4, c6, c7]
+    expectNoFilterQueryResult $
+      Just $
+        Map.fromList
+          [ (c3, CommitteeMemberState MemberResigned Active (Just 9) NoChangeExpected)
+          , (c4, CommitteeMemberState MemberNotAuthorized Active (Just 9) NoChangeExpected)
+          , (c6, CommitteeMemberState (MemberAuthorized (KeyHashObj hk6)) Active (Just 9) NoChangeExpected)
+          , (c7, CommitteeMemberState MemberNotAuthorized Expired (Just 7) NoChangeExpected)
+          ]
+  where
+    expectMembers :: Set.Set (KeyHash 'ColdCommitteeRole (EraCrypto era)) -> ImpTestM era ()
+    expectMembers expKhs = do
+      committee <-
+        getsNES $
+          nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . cgEnactStateL . ensCommitteeL
+      let members = Map.keysSet $ foldMap' committeeMembers committee
+      impAnn "Expecting committee members" $ members `shouldBe` Set.map KeyHashObj expKhs
+    expectQueryResult ckFilter hkFilter statusFilter expResult = do
```

Comment:
Could you add a type signature for this function with `HasCallStack` constraint, this will let the user know which usage of the function caused a test to fail.